gcore/gdal_priv.h

```diff
@@ -393,6 +393,7 @@ class CPL_DLL GDALDataset : public GDALMajorObject
 
     void RasterInitialize(int, int);
     void SetBand(int, GDALRasterBand *);
+    void SetBand(int nNewBand, std::unique_ptr<GDALRasterBand> &&poBand);
```

Comment:
FWIW, the ISO guideline would be to take the `unique_ptr` by value instead of rvalue reference: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f18-for-will-move-from-parameters-pass-by-x-and-stdmove-the-parameter

I have not done this in GEOS because I was not aware of the guideline when I started converting the API to unique_ptr, and I think it's important to keep it consistent. That said, I recall fixing one leak where a function took a unique_ptr by rvalue reference and then failed to move from it.

Reply:
interesting. Done